### DISCUSSION TITLE
Auto adjust sender stream based on consumers usage

### DISCUSSION THREAD

**Discussion by giangndm** [8xFF/atm0s-media-server#364] [Ideas]
For optimizing resource usage in sender side, we should have some thing like auto reduce or turn-off sender stream.

The ideally logic is:

- No one watch the stream => disable send
- Some other watching with condition: max layers, max allocated bitrate =>  don't send waste data which don't used by any consumer

**Comment by jibon57** [comment]
Thank you. From client side there should have option to disable/pause stream. This could be done either by a dedicated option or simple video pause. If option done by video pause then there have possibility user may play again. So, having a dedicated option will be better. From JS side can be use [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/observe) to determine the visibility of the video and pause when necessary.

**Reply by giangndm** [reply]
@jibon57 I think we can add something like [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/observe) with ts-sdk attach/detach. or pause/resume when it complete (https://github.com/8xFF/atm0s-media-sdk-ts/issues/20).
